src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -2,38 +2,54 @@
 
 public class IntLinkedList {
 
-	Node last;
-	int i;
+	private static final int EMPTY_LIST = -1;
+	private static final Node LIST_END = null;
+	private Node last;
+	private int i;
+
+	public Node getLast() {
+		return this.last;
+	}
+
+	public void setLast(Node last) {
+		this.last = last;
+	}
+
+	public int getI() {
+		return this.i;
+	}
 
 	public void push(int i) {
-		if (last == null)
-			last = new Node(i);
+		if (getLast() == LIST_END){
+			setLast(new Node(i));
+		}
 		else {
-			last.next = new Node(i);
-			last.next.prev = last;
-			last = last.next;
+			getLast().setNext(new Node(i));
+			getLast().getNext().setPrev(getLast());
+			setLast(getLast().getNext());
```

Comment:
This is an exaggeration. Using private fields internally (last field) usually does not require encapsulation.